### PR TITLE
Add AllocationStrategy to EMR instance fleet configuration

### PR DESCRIPTION
**Pull request by flamand-gh**
Added ability to add AllocationStrategy to the EMR instance fleet configuration. This change also adds OnDemandSpecification to InstanceFleetProvisioningSpecifications

### CODE REVIEW

File: troposphere/emr.py [hunk 1]
```diff
@@ -259,18 +259,46 @@ class InstanceGroupConfigProperty(AWSProperty):
...
+    }
+
+    def validate(self):
+        valid_values = [ 'lowest-price' ]
+
+        allocation_strategy = self.properties.get('AllocationStrategy', None)
+
+        if allocation_strategy not in valid_values:
+            raise ValueError(
+                'Only lowest-price is a valid AllocationStrategy'
```
Comment (PatMyron) [inline troposphere/emr.py:274]: should have more general messages so they don't have to be changed when `valid_values` expands

File: troposphere/emr.py [hunk 2]
```diff
@@ -259,18 +259,46 @@ class InstanceGroupConfigProperty(AWSProperty):
...
 
+    def validate(self):
+        if 'AllocationStrategy' in self.properties:
+            valid_values = [ 'capacity-optimized' ]
+
+            allocation_strategy = self.properties.get('AllocationStrategy', None)
+
+            if allocation_strategy not in valid_values:
+                raise ValueError(
+                    'Only capacity-optimized is a valid AllocationStrategy'
```
Comment (PatMyron) [inline troposphere/emr.py:294]: should have more general messages so they don't have to be changed when `valid_values` expands

File: troposphere/emr.py (hunk 1 above)
Comment (flamand-gh) [reply troposphere/emr.py]: Good remark. I changed the error message. It now also prints the provided invalid AllocationStrategy.

**Comment by flamand-gh** [comment]
Hi all, is there anything I can do to get this merged and released as soon as possible? Using AllocationStrategy in our EMR clusters is a suggestion from AWS to have a bigger chance of getting the required resources during Black Friday and Christmas holidays.

**Comment by benbridts** [comment]
@flamand-gh as a quick workaround, you can install troposphere from a (your) github branch too

```
pip install git+https://github.com/flamand-gh/troposphere.git@master
```

**Comment by flamand-gh** [comment]
> @flamand-gh as a quick workaround, you can install troposphere from a (your) github branch too
> 
> ```
> pip install git+https://github.com/flamand-gh/troposphere.git@master
> ```

Thanks for the suggestion! I'll try that.

**Comment by markpeek** [comment]
Thanks!

**Pull request merged**